airbyte-integrations/connectors/destination-keen/src/main/java/io/airbyte/integrations/destination/keen/KeenRecordsConsumer.java

```diff
@@ -0,0 +1,134 @@
+import io.airbyte.protocol.models.DestinationSyncMode;
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class KeenRecordsConsumer extends FailureTrackingAirbyteMessageConsumer {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(KeenRecordsConsumer.class);
+
+  private final JsonNode config;
+  private final ConfiguredAirbyteCatalog catalog;
+  private final Consumer<AirbyteMessage> outputRecordCollector;
+
+  private KeenTimestampService timestampService;
+  private String projectId;
+  private String apiKey;
+  private boolean timestampInferenceEnabled;
+  private KafkaProducer<String, String> kafkaProducer;
+  private AirbyteMessage lastStateMessage;
+  private Set<String> streamNames;
+
+  public KeenRecordsConsumer(JsonNode config,
+                             ConfiguredAirbyteCatalog catalog,
+                             Consumer<AirbyteMessage> outputRecordCollector) {
+    this.config = config;
+    this.catalog = catalog;
+    this.outputRecordCollector = outputRecordCollector;
+    this.kafkaProducer = null;
+    this.streamNames = Set.of();
+    this.lastStateMessage = null;
+    LOGGER.info("initializing consumer.");
+  }
+
+  @Override
+  protected void startTracked() throws IOException, InterruptedException {
+    projectId = config.get(CONFIG_PROJECT_ID).textValue();
+    apiKey = config.get(CONFIG_API_KEY).textValue();
+    timestampInferenceEnabled = Optional.ofNullable(config.get(INFER_TIMESTAMP))
+        .map(JsonNode::booleanValue)
+        .orElse(true);
+    this.kafkaProducer = KeenDestination.KafkaProducerFactory.create(projectId, apiKey);
+    this.streamNames = getStrippedStreamNames();
+    this.timestampService = new KeenTimestampService(this.catalog, timestampInferenceEnabled);
+    eraseStreams();
+  }
+
+  @Override
+  protected void acceptTracked(AirbyteMessage msg) {
+    if (msg.getType() == Type.STATE) {
+      lastStateMessage = msg;
+      outputRecordCollector.accept(lastStateMessage);
+      return;
+    } else if (msg.getType() != Type.RECORD) {
+      return;
+    }
+
+    final String streamName = getStreamName(msg.getRecord());
+    JsonNode data = this.timestampService.injectTimestamp(msg);
+
+    kafkaProducer.send(new ProducerRecord<>(streamName, data.toString()));
+  }
+
+  private Set<String> getStrippedStreamNames() {
+    return catalog.getStreams()
+        .stream()
+        .map(ConfiguredAirbyteStream::getStream)
+        .map(AirbyteStream::getName)
+        .map(KeenCharactersStripper::stripSpecialCharactersFromStreamName)
+        .collect(Collectors.toSet());
+  }
+
+  private void eraseStreams() throws IOException, InterruptedException {
```

Comment:
small thing: I suggest renaming to `eraseOverwriteStreams`

Reply:
done